metadata-ingestion/tests/integration/snowflake/test_snowflake_queries.py

```diff
@@ -22,3 +22,49 @@ def test_source_close_cleans_tmp(snowflake_connect, tmp_path):
         # This closes QueriesExtractor which in turn closes SqlParsingAggregator
         source.close()
         assert len(os.listdir(tmp_path)) == 0
+
+
+@patch("snowflake.connector.connect")
+def test_user_identifiers_email_as_identifier(snowflake_connect, tmp_path):
+    with patch("tempfile.tempdir", str(tmp_path)):
+        source = SnowflakeQueriesSource.create(
+            {
+                "connection": {
+                    "account_id": "ABC12345.ap-south-1.aws",
+                    "username": "TST_USR",
+                    "password": "TST_PWD",
+                },
+                "email_as_user_identifier": True,
+                "email_domain": "example.com",
```

Comment:
do we miss the case below? it that makes sense

```
email_as_user_identifier: True
email_domain: None
```

Reply:
That's true -> I'll add that to test -> but in this case -  we'd do best effort and then if email still can't be formed then use username for urn.